config/configuration.go

```
@@ -0,0 +1,125 @@
+	Password string `yaml:Password`
+}
+
+type IQConfig struct {
+	Server   string `yaml:"Server"`
+	Username string `yaml:"Username"`
+	Token    string `yaml:"Token"`
+}
+
+type Config struct {
+	logger *logrus.Logger
+	OSSIndexConfig OSSIConfig
+	IQConfig IQConfig
+}
+
+func New(logger *logrus.Logger) *Config {
+	return &Config{logger: logger}
+}
+
+func (c *Config) CreateOrReadConfigFile() {
+
+	c.createDirectory(types.IQServerDirName)
+	c.createDirectory(types.OssIndexDirName)
+
+	if !fileExists(c.getIQConfig()) {
+		c.writeDefaultIQConfig()
+	}
+	if !fileExists(c.getOssiConfig()) {
+		c.writeDefaultOssiConfig()
+	}
+
+	c.readConfig()
+}
+
+func (c Config) createDirectory(directory string) {
+	home, _ := os.UserHomeDir()
+	fileDirectory := filepath.Join(home, directory)
+	err := os.MkdirAll(fileDirectory, os.ModePerm)
+	if err != nil {
+		logger.Error("Couldn't create config directory: " + fileDirectory)
+	}
+}
+
+//Gets the default location for the config file
+func (c Config) getIQConfig() string {
+	home, _ := os.UserHomeDir()
+	return filepath.Join(home,types.IQServerDirName, types.IQServerConfigFileName)
+
+}
+
+func (c Config) getOssiConfig() string {
+	home, _ := os.UserHomeDir()
+	return filepath.Join(home,types.OssIndexDirName, types.OssIndexConfigFileName)
+}
+
+func (c *Config) readConfig() {
+	iqBytes, err := ioutil.ReadFile(c.getIQConfig())
+	if err != nil {
+		c.logger.Error(err)
+	}
+	iqConfig := IQConfig{}
+	_ = yaml.Unmarshal(iqBytes, &iqConfig)
+
+	ossiBytes, err := ioutil.ReadFile(c.getOssiConfig())
+	if err != nil {
+		c.logger.Error(err)
+	}
+	ossiConfig := OSSIConfig{}
+	_ = yaml.Unmarshal(ossiBytes, &ossiConfig)
+
+	c.OSSIndexConfig = ossiConfig
+	c.IQConfig = iqConfig
+}
+
+
+func (c Config) writeDefaultOssiConfig() {
+	ossiConfig, _ := yaml.Marshal(OSSIConfig{})
+	err := ioutil.WriteFile(c.getOssiConfig(), ossiConfig, 0644)
+	if err != nil {
+		c.logger.WithField("configFile", c.getOssiConfig()).Error("Could not create OSSIndeConfig.")
```

You ignore the error here, which can be fine, but it might help someone later (maybe it can't write to a location due to perms or something like that)